apps/web-client/src/lib/stores/inventory/table_content.ts

```diff
@@ -1,45 +1,85 @@
 
-import type { BookStore, DisplayRow, PaginationData } from '$lib/types/inventory';
+import type { PaginationData, DisplayRow } from '$lib/types/inventory';
 
 import { readableFromStream } from '$lib/utils/streams';
+import { from, map, Observable, switchMap } from 'rxjs';
+import type { DebugCtx } from '@librocco/shared/dist/debugger';
 
 interface CreateDisplayEntriesStore {
 	(
+		db: DatabaseInterface<WarehouseInterface<NoteInterface<object>, object>, NoteInterface<object>>,
 		entity: NoteInterface | WarehouseInterface | undefined,
 		currentPageStore: Readable<number>,
-		bookStore: Readable<BookStore>,
 		ctx: debug.DebugCtx
 	): Readable<DisplayRow[]>;
 }
+
+interface CreateDisplayRowStream {
+	(
+		db: DatabaseInterface<WarehouseInterface<NoteInterface<object>, object>, NoteInterface<object>>,
+		entity: NoteInterface<object> | WarehouseInterface<NoteInterface<object>, object>,
+		ctx: DebugCtx
+	): Observable<DisplayRow[]>;
+}
+
+const createDisplayRowStream: CreateDisplayRowStream = (db, entity, ctx) => {
+	const fullTableRow = entity?.stream(ctx).entries.pipe(
+		switchMap((valueFromEntryStream) => {
+			// map entry to just isbns
+			const isbns = valueFromEntryStream.map((entry) => entry.isbn);
+
+			// return array of merged values of books and volume stock client
+			return from(db.getBooks(isbns)).pipe(
+				map((booksFromDb) => {
+					const booksMap = new Map();
+					booksFromDb.forEach((book) => book && booksMap.set(book.isbn, book));
+
+					// for each value from entry stream (volume stock client)
+					return valueFromEntryStream.map((individualEntry) => {
+						// find the corresponding book in booksFromDb
+						const bookData = booksMap.get(individualEntry.isbn);
+						// account for books not found
+						if (!bookData) return individualEntry;
+						// return bookData + entry value
+						const fullTableElement = { ...bookData, ...individualEntry };
+						return fullTableElement;
+					});
+				})
+			);
+		})
+	);
+
+	return fullTableRow;
+};
+
 /**
  * Creates a store that streams the entries to be displayed in the table, with respect to the content in the db and the current page (set by pagination element).
+ * @param db current context database
  * @param entity a note or warehouse interface
  * @param currentPageStore a store that containing the current page index
- * @param bookStore a store with book data - used to extend each entry from the entity store with the rest of the book data
+ * @param ctx debug context
  * @returns
  */
-export const createDisplayEntriesStore: CreateDisplayEntriesStore = (entity, currentPageStore, bookStore, ctx) => {
-	const entriesStore = readableFromStream(entity?.stream(ctx).entries, [], ctx);
+export const createDisplayEntriesStore: CreateDisplayEntriesStore = (db, entity, currentPageStore, ctx) => {
+	const fullTableRowStream = createDisplayRowStream(db, entity, ctx);
+	const displayRowStore = readableFromStream(fullTableRowStream, [], ctx);
```

Comment:
Let's call this `entriesStore` like before, because here you're:
1. Creating `displayRows`
2. Creating `displayEntries` from `displayRows`...

The former aren't displayed (they're used to create the displayed ones) and, if any should be `displayRows` (implying they're displayed in the UI as table rows), those should be the latter.